src/components/weather/weather.jsx

```diff
@@ -0,0 +1,49 @@
+import { useEffect, useState } from "react";
+import { getWeather } from "../../services";
+import "./weather.css";
+
+export const Weather = () => {
+  const [weather, setWeather] = useState(null);
+  const [loc, setLoc] = useState({ lat: "", lon: "" });
+
+  useEffect(() => {
+    const options = {
+      enableHighAccuracy: true,
+      timeout: 5000,
+      maximumAge: 0,
+    };
+
+    const success = (pos) => {
+      const crd = pos.coords;
+      setLoc((loc) => ({
+        ...loc,
+        lat: crd.latitude,
+        lon: crd.longitude,
+      }));
+    };
+
+    const error = () => {
+      console.log("Sorry there is an error");
+    };
+    navigator.geolocation.getCurrentPosition(success, error, options);
+  }, []);
+
+  useEffect(() => {
+    getWeather(loc?.lat, loc?.lon, setWeather);
```

Comment:
What is the optional parameter being used for ?

Reply:
initially value of lat and lon is empty string